Remotion/Web/UnitTests/Core/ContentSecurityPolicy/CspEnabledHtmlTextWriterTests.cs

```diff
@@ -0,0 +1,205 @@
+    public void AddAttribute_WithSupportedEventType_NonceValueIsRendered (string eventType)
+    {
+      _randomNumberGeneratorMock
+          .Setup(m => m.GenerateAlphaNumericNonce())
+          .Returns("eventTargetID")
+          .Verifiable();
+
+      _clientScriptMock
+          .Setup(m => m.RegisterStartupScriptBlock(_pageMock.Object, typeof(CspEnabledHtmlTextWriter), It.IsAny<string>(), It.IsAny<string>()))
+          .Verifiable();
+
+
+      _writer.AddAttribute(eventType, "console.info('test');");
+      _writer.RenderBeginTag(HtmlTextWriterTag.Button);
+      _writer.RenderEndTag();
+
+      var document = _htmlHelper.GetResultDocument();
+      var element = _htmlHelper.GetAssertedChildElement(document, "button", 0);
+      _htmlHelper.AssertAttribute(element, "data-inline-event-target", "eventTargetID");
+
+
+      _clientScriptMock.Verify();
+      _randomNumberGeneratorMock.Verify(m => m.GenerateAlphaNumericNonce(), Times.Once());
+    }
+
+    [Test]
+    public void AddAttribute_WithAttributeOtherThanEventType_NonceValueIsNotRendered ()
+    {
+      _writer.AddAttribute("style", "color:white;");
+      _writer.RenderBeginTag(HtmlTextWriterTag.Button);
+      _writer.RenderEndTag();
+
+      var document = _htmlHelper.GetResultDocument();
+      var element = _htmlHelper.GetAssertedChildElement(document, "button", 0);
+      _htmlHelper.AssertNoAttribute(element, "data-inline-event-target");
+
+      _clientScriptMock.Verify();
+    }
+
+    [Test]
+    public void AddAttribute_WithMoreThanOneAttribute_NonceValueIsRendered ()
+    {
+      _randomNumberGeneratorMock
+          .Setup(m => m.GenerateAlphaNumericNonce())
+          .Returns("eventTargetID")
+          .Verifiable();
+
+      _clientScriptMock
+          .Setup(m => m.RegisterStartupScriptBlock(_pageMock.Object, typeof(CspEnabledHtmlTextWriter), It.IsAny<string>(), It.IsAny<string>()))
+          .Verifiable();
+
+      _writer.AddAttribute("onclick", "console.info('test1');");
+      _writer.AddAttribute("onchange", "console.info('test2');");
+      _writer.RenderBeginTag(HtmlTextWriterTag.Button);
+      _writer.RenderEndTag();
+
+      var document = _htmlHelper.GetResultDocument();
+      var element = _htmlHelper.GetAssertedChildElement(document, "button", 0);
+      _htmlHelper.AssertAttribute(element, "data-inline-event-target", "eventTargetID");
+
+      _clientScriptMock.Verify();
+      _randomNumberGeneratorMock.Verify(m => m.GenerateAlphaNumericNonce(), Times.Once());
+    }
+
+    [Test]
+    public void AddAttribute_WithMoreThanOneTag_NonceValueIsRendered ()
+    {
+      _randomNumberGeneratorMock
+          .SetupSequence(m => m.GenerateAlphaNumericNonce())
+          .Returns("eventTargetID1")
+          .Returns("eventTargetID2");
+
+      _clientScriptMock
+          .Setup(m => m.RegisterStartupScriptBlock(_pageMock.Object, typeof(CspEnabledHtmlTextWriter), It.IsAny<string>(), It.IsAny<string>()))
+          .Verifiable();
```

Comment:
There should be qualified expections set up. Two calls, two eventIDs, two script-strings.

consdering the rest of the setup, we might actually want to move this to line 179 and use only two verify statements with the expected parameters instead of setting it up before the calls. The API returns void and by asserting it at the end / next to the htmlheler-assertAttribute calls, we can better establish the relationship